### ISSUE TITLE
Face Verification 

### ISSUE THREAD

**Issue by sanjanajainsj** [happynear/FaceVerification#42]
Hello Sir,
Thank you for the models and codes you have provided. We are planning to train a face verification model on the CASIA-WebFace dataset from scratch using the "mnist_siamese_train_test.prototxt" model you provided. I have gone through the papers you specified in the previous issues already. How are you generating the training pairs and also how are you labeling them? Thank you in advance.

**Comment by happynear** [comment]
Don't use that. I never got success in training the "mnist_siamese_train_test.prototxt" model.

I suggest you to download CASIA-Webface, clean the dataset with the list I provided, and train it using center-face (https://github.com/ydwen/caffe-face). If everything works fine, you will get ~99% on LFW.

**Comment by sanjanajainsj** [comment]
Hello Sir,
Thank you for your reply. I have a few questions:
1) I have already downloaded the washed-up Casia-WebFace dataset. Which list should I be using to clean the dataset? 
2) I was planning to train a Siamese model for face verification and the model from caffe-face (https://github.com/ydwen/caffe-face) seems to be a single-cnn model. If possible, could you suggest a Siamese network for face verification.
3) Also, I was going through one of the issues (https://github.com/happynear/FaceVerification/issues/22), where you provided the caffemodel, meanfile and deploy.prototxt. However, the deploy.prototxt file ends at the dropout layer, I am confused how do we get an output as we do not have any output layer like softmax, etc.

**Comment by happynear** [comment]
- If you downloaded the washed dataset, there is no need to use the list again.

- Currently I don't see successful open-source Siamese network for face verification, just use center-face. If you see the DeepID2 paper, you will find that optimizing the inter-class variance only could already lead to very good performance.

- deploy.prototxt is for extracting features, you may just add an innerproduct layer and a softmax layer afer it to train a model.

**Comment by sanjanajainsj** [comment]
Hello Sir,
Thank you for your reply. I have a few questions regarding this:

1. I tried to finetune your model with another dataset of unaligned images. I am using "CASIA_train_test.prototxt", "mnist_siamese_solver.prototxt" and the model you provided in https://github.com/happynear/FaceVerification/issues/22 . I tried with 100,000 iterations and my loss did not decrease since the start. I am getting a testing loss of 4.72074 and testing accuracy of 0.0072. I am labeling the dataset like this:

/home/.../dataset/subject1/img1 1 
/home/.../dataset/subject1/img2 1 
/home/.../dataset/subject2/img1 2 
/home/.../dataset/subject2/img2 2
...

Is this the correct way to label the dataset? 

2. I am planning to train the models for both your cnn and center-face from scratch. How should the labeling be done?

**Comment by happynear** [comment]
Your labels are good, both for CASIA_train_test.prototxt and center-face.

**Comment by sanjanajainsj** [comment]
Hello Sir,
What could be the reason for the loss not decreasing and accuracy of 0.0072? I tried training Casia from scratch and I am getting similar results. I tried for 70k iterations and it had loss of 4.26 and accuracy of 0.0072.

**Comment by baby313** [comment]
Hi Sir,
I followed your suggestion and center-face. However, the memory cost is too much to train 256/batch. As a result, I change it to 128, but the center_loss did not decrease :( I intended to do face matching on phone and realize that center_face may be too heavy (Speed and Model Size) to do so. Do you have any suggestion for face matching on phone? DeepID2 is the best choice I have found currently and do you know other better options?

**Comment by sidgan** [comment]
@sanjanajainsj Could you provide a link to the cleaned dataset.

**Comment by sanjanajainsj** [comment]
@sidgan I found the cleaned dataset on this link: http://pan.baidu.com/s/1jIqBIcu and the password is eb7h

**Comment by sidgan** [comment]
For me this page does not translate to English, I tried a lot of online translation tools, and I don't know Chinese. Additionally, the Baidu installer only has Chinese. So, for me that isn't helpful either. Do you know of a workaround to this?

**Comment by yao5461** [comment]
@sanjanajainsj Hi, the link has been invalid, any other accessible links? Thanks!
